x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/CompositeRolesStore.java

```diff
@@ -532,6 +532,7 @@ public static void buildRoleFromDescriptors(
             privilegeStore.getPrivileges(
                 applicationNames,
                 applicationPrivilegeNames,
+                false, // TODO revisit if we should also wait for an available security index here
```

Comment:
This is called during role building, for roles that have application privileges defined -- I'm opting to omit the wait for now since it's wider impact than originally discussed. The target fix of this PR still holds (largely) since Get Privileges are called with Kibana system role which does not have app privileges.

All that said, esp. if we narrow the wait condition to "index is initializing" I think it's acceptable (and desirable) to also wait here.